tools/mason/MasonExternal.chpl

```diff
@@ -114,8 +113,14 @@ proc masonExternal(args: [] string) {
         const spackLatestBranch = ' --branch ' + spackBranch + ' ';
         const status = cloneSpackRepository(spackLatestBranch, SPACK_ROOT);
         if isDir(MASON_HOME + '/spack-registry') then generateYAML();
-        if status != 0 then throw new owned MasonError("Spack installation failed.");
+        if status != 0 then
+          throw new owned MasonError("Spack installation failed.");
       }
+      // check that after all this, the version of spack is as we expect it
+      if getSpackVersion != spackVersion then
```

Comment:
I'm worried that this still might be too strict, but let's see how things play out.

Reply:
I considered making it just a compatibility check, but decided against it because `mason external --setup` doesn't have a `force` option. So if a user somehow had `0.16.4`, and we required `0.16.3`, if they wanted to troubleshoot by first aligning the spack versions, there'd be no way to do that using mason.